active/0109-environment-selection.md

```diff
@@ -0,0 +1,90 @@
+# Environment Selection
+* Start Date: 2017-07-24
+* RFC PR: (leave this empty)
+
+# Summary
+
+The goal of this RFC is to justify and solidify the mechanism that ember-cli uses to determine the environment that any given command will be ran under. 
+
+# Motivation
+
+We have received feedback from our direct users, developer relations folks, and benchmark maintainers suggesting that our current system for selecting the environment does not set our users up for success.
+
+This RFC aims to ensure that users of ember-cli get the correct environment when specified (via `NODE_ENV`, `EMBER_ENV`, or `--environment` command line flags) and when using the default.  
+
+# Terminology
+
+* Command line specified environment -- When the command line invocation includes a specified `environment`  (either `--environment=xyz` or uses the environment shortcut aliases (e.g. `-prod` or `-dev`).
+* ember-cli specific process environment variable specified environment -- When the shell that invokes the command has `process.env.EMBER_ENV` set.
+* General purpose process environment variable specified environment -- When the shell that invokes the command has `process.env.NODE_ENV` set.
+* Test command -- A command used to run tests (e.g. `ember test` or `ember test --server`).
+* Interactive command -- A command which is not intended to "complete". These commands commonly perform rebuilds and/or re-run tests (i.e. `ember build --watch`, `ember serve`). 
+* Non-interactive command -- A command intended to be ran to completion (e.g. `ember build` or `ember test`).
+
+# Detailed design
+
+The following mechanism will be used to select the environment by each ember-cli command (in order):
+
+1. Command line specified environment (e.g. `ember serve --environment production`)
+2. ember-cli specific process environment variable specified environment (e.g. `EMBER_ENV=production ember serve`)
+3.  General purpose process environment variable specified environment (e.g. `NODE_ENV=production ember serve`)
+4. No environment was specified, default based on command type:
+	* Test commands -- Default to `test` environment.
```

Comment:
Based on the definitions above, "test commands" could also be considered "interactive" (`ember test --server`) or "non-interactive" (`ember test`), therefore I think we should specify an order of precedence here as well.